tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -4,18 +4,94 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"time"
 
 	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
 	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	nodestat "github.com/rancher/rancher/tests/framework/extensions/nodes"
+	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
 )
 
+const (
+	active = "active"
+)
+
+// MatchRoleToConfig matches the role of machinePools to the nodeRoles.
+func MatchRoleToConfig(nodeRoles NodeRoles, machinePools []apisV1.RKEMachinePool) (int, int32) {
```

Comment:
Could this get a new name, since this is just for machinePools and not rke1 cluster types? Also this is similar to what I had in scale_replace_test.go, maybe we could consolidate